Source/Core/Core/HW/MemmapFunctions.cpp

```diff
@@ -460,6 +466,44 @@ void WriteUnchecked_U32(const u32 var, const u32 address)
 	WriteToHardware<FLAG_NO_EXCEPTION, u32>(address, var);
 }
 
+void DMA_LCToMemory(const u32 memAddr, const u32 cacheAddr, const u32 numBlocks)
+{
+	// TODO: It's not completely clear this is the right spot for this code;
+	// what would happen if, for example, the DVD drive tried to write to the EFB?
+	// TODO This is terribly slow.
+	// Avatar: The Last Airbender (GC) uses this for videos.
+	if ((memAddr & 0x0F000000) == 0x08000000)
+	{
+		for (u32 i = 0; i < 32 * numBlocks; i+=4)
+		{
+			u32 data = bswap(*(u32*)(Memory::m_pL1Cache + ((cacheAddr + i) & 0x3FFFF)));
+			EFB_Write(data, memAddr + i);
+		}
+		return;
+	}
+
+	const u8* src = Memory::m_pL1Cache + (cacheAddr & 0x3FFFF);
+	u8* dst = Memory::GetPointer(memAddr);
```

Comment:
I'll add an explicit check.  Not sure what we're supposed to do in that case, though... isn't MMIO sensitive to the size of writes?